src/libs/sfdk/signingutils.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "asynchronous.h"
+
+#include "sfdkglobal.h"
+#include <utils/fileutils.h>
+#include <QObject>
+
+namespace Sfdk {
+
+class BuildEngine;
+
+struct GpgKeyInfo
+{
+    QString name;
+    QString fingerprint;
+};
+
+typedef QVector<GpgKeyInfo> GpgKeyInfoList;
+
+class SigningUtilsPrivate;
+class SFDK_EXPORT SigningUtils : public QObject
+{
+    Q_OBJECT
+public:
+    explicit SigningUtils(QObject *parent = nullptr);
+    virtual ~SigningUtils();
+
+    void availableKeys(const QObject *context, const Functor<bool, const GpgKeyInfoList &> &functor);
+    void exportSecretKey(const QString &fingerprint,
+                         const Utils::FilePath &keyPassFile,
+                         const Utils::FilePath &outputDir,
+                         const QObject *context,
+                         const Functor<bool> &functor);
+    void exportPublicKey(const QString &fingerprint,
+                         const QObject *context,
+                         const Functor<bool, const QByteArray &> &functor);
+    QString error() const;
+
+protected:
+    SigningUtils(std::unique_ptr<SigningUtilsPrivate> &&d, QObject *parent);
+    std::unique_ptr<SigningUtilsPrivate> d_ptr;
+
+private:
+    Q_DECLARE_PRIVATE(SigningUtils);
+    QString m_error;
+};
```

Comment:
If we take out Sfdk::Device::importGpgKey we lose the ability to set the key in MerHardwareDeviceWizardPackageKeyDeploymentPage when adding a device. I think this is more convenient than a separate step and we need to keep this feature

Reply:
I am not sure I follow what you mean, but as I see it most of the code that is now part of `PackageKeyDeployerPrivate::deploy` would be moved under `Sfdk::Device::importGpgKey(const QString &id, const QObject *context, const Functor<bool, QString> &functor)` which can use `SigningTools` internally.

`MerHardwareDeviceWizardPackageKeyDeploymentPage::deployKey()` would than directly do

```
const QString id = gpgKeyInfoList.at(m_keyComboBox->currentIndex()).fingerprint;
m_device->importGpgKey(id, this, [&](bool ok, const QString &errorString) {
    /* same as now */
})
```

Reply:
Now I see that `m_device` is a `MerHardwareDevice` and `Sfdk::HardwareDevice` does not exit yet at that point. - Is that what you meant?

This is not going to change easily :\ OK, let's create `static Sfdk::Device::importGpgKey(const QSsh::SshConnectionParameters &sshParameters, const QString &id, const QObject *context, const Functor<bool, QString> &functor)` as a workaround then and let `MerHardwareDeviceWizardPackageKeyDeploymentPage::deployKey()` do

```
const QString id = gpgKeyInfoList.at(m_keyComboBox->currentIndex()).fingerprint;
HardwareDevice::importGpgKey(m_device->sshParameters(), id, this,
        [&](bool ok, const QString &errorString) {
    /* same as now */
})
```

This will allow us to follow the long term objective of minimizing the amount of code in the Mer plugin and keeping as much as possible inside the Sfdk library. Please mark it with "TODO Make non-static".